### ISSUE TITLE
Cryptocurrency address associated with entity

### ISSUE THREAD

**Issue by TJKoury** [schemaorg/schemaorg#2964]
I am working on data standards for space, and am currently building an app to aid the community in passing data between consumers.

I need a field like the one I defined in this [ICD proposal extension](https://github.com/DigitalArsenal/spacedatastandards.org/blob/master/docs/schemas/ICD/ODM_FIELDS_DIGITALARSENAL.pdf) to capture a cryptocurrency address associated with a Person, Organization, Ground Site, Resident Space Object, etc.

Any thoughts from the community?  I would be happy to submit something if this has not been considered yet.

**Comment by WeaverStever** [comment]
As a person w limited experience in crypto, I think this might be a little premature as crypto addresses are still complicated and publishing in this manner may cause crypto to be destroyed by sending to an invalid address.  

Alternately, Coinbase has an on-exchange system where I only need to know the associated email address to send crypto to -- and what type of crypto I am sending -- all of the details are handled and there is no fee for sending from one Coinbase account another.  (Unstoppabledomains dot com appears to provide a similar service.)

I realize that Coinbase is a CEX and not a DEX, but the beauty of this approach is that the wallet address is not exposed to the internet.  My thinking here is, do I really want to expose to the public how much BTC is in a specific wallet and disclose that this wallet belongs to me?

**Comment by afomi** [comment]
Satoshi and best practices advocate against address reuse. 

[Section 10](https://bitcoin.org/bitcoin.pdf)

> a new key pair should be used for each transaction to keep them from being linked to a common owner.

As @WeaverStever mentioned, other types of schemes or aliases may emerge to meet this need, like [paymail](https://bsvalias.org/).

**Comment by TJKoury** [comment]
@afomi If your goal is anonymity, then yes.  My particular use case is to marry traditional PKI and cryptocurrency, put funds against private keys used in traditional roles.

It is part of a system of adversarial trust, the ability to increase trust that a key has not been compromised by sending funds to it and having those funds remain in place.

It’s easy to claim a certificate authority, web server, code-signing certificate, etc. is uncompromised.  Proving it is a whole other matter.

Consider [Netfilter](https://www.venafi.com/blog/microsoft-signed-rootkit-malware-spreads-through-gaming).  Microsoft still doesn’t know (or won’t say) if the review process was compromised or the signing key itself.  

Put $100 million in BTC against the private key, and if it doesn’t move then it’s a good bet that it is not compromised, game-theory wise.

**Comment by danbri** [comment]
I would encourage you to try to find an idiom that re-uses existing schema.org terms (e.g. Person, Organization, https://schema.org/DigitalDocument, https://schema.org/encodingFormat ...). There is so much churn and chaos in the "crypto"(i.e. cryptocurrency) scene right now I would be wary of baking something into Schema.org at this point.

@chaals may have opinions, I suspect

**Comment by TJKoury** [comment]
I understand not trying to tie something as permanent as Schema.org into some of the more volatile of the crypto community.  At the same time, Bitcoin pre-dates Schema.org, so including a reference for Bitcoin legacy addresses at least seems appropriate.

**Comment by WeaverStever** [comment]
Perhaps one of these can be adapted?
[InvestmentOrDeposit](https://schema.org/InvestmentOrDeposit)

**subtypes**
[BrokerageAccount](https://schema.org/BrokerageAccount)
[DepositAccount](https://schema.org/DepositAccount)
[InvestmentFund](https://schema.org/InvestmentFund)

**Comment by nerdfiles** [comment]
no

**Comment by TJKoury** [comment]
So I'm creating a vCard (v4.0) <-> Person converter in TypeScript, and I'm basically just adding an `ITEM{N}.X-ABRELATEDNAMES` field to store the pubic key, and another for the Bitcoin address.  Not sure which field to jam those in for the Person schema, willing to submit a pull, thoughts?

**Comment by TJKoury** [comment]
Here is what I have settled on (in TypeScript):

```
import { Person, ContactPoint } from "digital-arsenal-schema-dts";


export type cryptoKeyBase = {
    //hex publicKey
    key: string,
    keyAddress?: string,
    //https://github.com/satoshilabs/slips/blob/master/slip-0044.md
    keyType?: number
}

export interface cryptoKey extends cryptoKeyBase {
    "@type": "CryptoKey",
}

export type PersonPublicKey = Person & {
    key: cryptoKey | Array<cryptoKey>
}
```
Again, willing to submit a pull and do the work if this is something that you will consider.

**Comment by friedger** [comment]
> My thinking here is, do I really want to expose to the public how much BTC is in a specific wallet and disclose that this wallet belongs to me?

[netWorth](https://schema.org/netWorth) is property of Person.

When adding cryptocurrency addresses, then it should follow chain agnostic standard format, i.e. just a string in a special format.

Best property to be used is IMHO `owns`, e.g.

```json
{
  "@type": "Person",
  "familyName": "Nakamoto",
  "givenName": "Satoshi",
  "owns": [
      {"@type": "OwnershipInfo", "identifier": "bip122:000000000019d6689c085ae165831e93:12cbQLTFMXRnSzktFkuoG3eHoMeFtpTu3S"}
    ]
}
```

**Comment by chrisspradling1980** [comment]
Exactly. Mining upon completion exposes the block chain verification
anyways. It is not that difficult to obtain a wallet after an app has
identified as inactive. Wouldn't it be possible to obtain it after this has
occurred?

Good point!
